### PR TITLE
drivers: sensor: Update drivers to use devicetree Kconfig symbol

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Comment by bbilas** [comment]
@galak Are you gonna also remove the unnecessary Kconfig symbols from samples? For example https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/sensor/ina219/prj.conf#L5=

**Comment by galak** [comment]
> @galak Are you gonna also remove the unnecessary Kconfig symbols from samples? For example https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/sensor/ina219/prj.conf#L5=

Not in this PR.  Happy to have help from others on that.

**Review by bbilas — DISMISSED** [review]

File: drivers/sensor/adxl345/Kconfig [hunk 1]
```diff
@@ -7,6 +7,7 @@ config ADXL345
 	bool "ADXL345 Three Axis accelerometer"
 	default y
 	depends on DT_HAS_ADI_ADXL345_ENABLED
-	depends on I2C || SPI
+	depends on (I2C && $(dt_compat_on_bus,$(DT_COMPAT_ADI_ADXL345),i2c)) || \
+		   (SPI && $(dt_compat_on_bus,$(DT_COMPAT_ADI_ADXL345),spi))
```
Comment (MaureenHelm) [inline drivers/sensor/adxl345/Kconfig:11]: I don't quite understand this logic. The sensor has to be on one type of bus or the other, so isn't it redundant to qualify with `dt_compat_on_bus` now that it depends on `DT_HAS_ADI_ADXL345_ENABLED`?

Comment (galak) [reply drivers/sensor/adxl345/Kconfig:11]: No, there was a case that the sensor was on I2C and enabled, but CONFIG_I2C=n and CONFIG_SPI=y.  So `DT_HAS_ADI_ADXL345_ENABLED` will be 'y', so we need to qualify such a case (which is why that mess of logic).

**Review by MaureenHelm — APPROVED** [review]

**Review by bbilas — APPROVED** [review]

**Pull request merged**